app/src/main/java/com/example/tanya/godeltechchallengeandroid/di/module/StartModule.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.tanya.godeltechchallengeandroid.di.module
+
+import com.example.tanya.godeltechchallengeandroid.ui.start.StartContract
+import com.example.tanya.godeltechchallengeandroid.ui.start.StartPresenter
+import com.example.tanya.godeltechchallengeandroid.util.Prefs
+import dagger.Module
+import dagger.Provides
+
+
+@Module
+class StartModule {
+
+    @Provides
+    fun providePresenter(prefs: Prefs): StartContract.Presenter {
```

Comment:
This is incorrect! The StartPresenter instance creation is already done by the code below:
`@Inject constructor(private var prefs: SharedPreferences) {`

Here you have to use the Binds annotation if you want to cast the concrete presenter implementation to its contract and inject the contract to an activity.